net/http/client.cpp

```diff
@@ -73,6 +73,7 @@ ISocketStream* PooledDialer::dial(std::string_view host, uint16_t port, bool sec
     if (secure) {
         tlssock->timeout(timeout);
         sock = tlssock->connect(ep);
+        tls_socket_set_hostname(sock, strhost.c_str());
```

Comment:
Make SNI check optional seems to be better. User should able to set enable or disable SNI check.

Add a method to set by `Operation` or `HTTPClient` should be a good idea.

Reply:
Is there a reason not to use SNI?

Go's HTTP client, curl, and all modern browsers use SNI by default (which means docker and containerd do as well).

Reply:
@Coldwings Is there any drawbacks to always use SNI? Compatibility with older versions of OpenSSL?

Reply:
> @Coldwings Is there any drawbacks to always use SNI? Compatibility with older versions of OpenSSL?

SNI extension and `SSL_set_tlsext_host_name` API is provided in OpenSSL 1.1.x but not exists in OpenSSL 1.0.2 versions. SNI feature is still supported in OpenSSL 1.0.2 but should be set by `SSL_ctrl`.

> Is there a reason not to use SNI all the time?
> 
> Go's HTTP client, curl, and all modern browsers use SNI by default (which means docker and containerd do as well).

Enable SNI by default is good. I just think that as a common library, provide a option to disable SNI might be useful when user needs.